### DISCUSSION TITLE
Can we update classes/props based on function conditions?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#1180] [Q&A] [answered]
### Question

Hi All,

I am trying to change the styling of some elements based on some conditions. So let's say for ex:
```python
toggle = ui.switch('Switch', on_change=lambda: change_label())
label = ui.label('Switch is off').classes('text-xs')

def change_label():
    if toggle.value:
        label.text = 'Switch is ON'
        label.classes('text-lg')
    else :
        label.text = 'Switch is OFF'
        label.classes('text-xs')
```

The idea is I want to control the label based on if the switch is turned on or off. I realized the text content is getting updated every time I press the switch. But the 2nd line, where I am trying to update the ```label.classes('text-xs')``` does not work properly.
It works the first time then stops.  

Is there a way I can change/update the classes/props like this based on conditions, so that I can control the styling?
Many thanks for the help.

Regards,
Anindya

**Comment by falkoschindler** [comment, marked as answer]
Hi @Anindya088!
The problem is, that you keep _adding_ classes. So eventually the label has both classes "text-lg" and "text-xs".
You have multiple options to solve this issue:
1. Replace all classes with the one you need:
    ```py
    label.classes(replace='text-lg')
    ```
2. Remove the class you don't need and add a new one:
    ```py
    label.classes(remove='text-xs', add='text-lg')
    ```
3. Use CSS instead:
    ```py
    label.style('font-size: 18px')
    ```
    This works because the `style` method keeps track of a style dictionary. Setting "font-size" to a different value will overwrite the dictionary entry, thus replaces the previous value.

**Reply by Anindya088** [reply]
Hi @falkoschindler ,

Oh now I understand! This makes more sense now about how to control the classes.

I thought adding a ```text-lg``` automatically removes the previous ones, did not realize they are getting added on top of previous ones.
I have a bunch of other classes also, so I think your option 2 is the best solution. 

Thanks a lot again for this,
Anindya
